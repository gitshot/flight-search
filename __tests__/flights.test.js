import React from 'react';
import renderer from 'react-test-renderer';

import Flights from '../pages/flights.js';

const flights = [
  {
    node: {
      id: 'RmxpZ2h0OjM0MzUzMjQ1Ng==',
      departure: {
        time: '2018-03-06T20:10:00.000Z',
        localTime: '2018-03-06T21:10:00.000Z'
      },
      arrival: {
        time: '2018-03-06T22:35:00.000Z',
        localTime: '2018-03-06T23:35:00.000Z'
      },
      legs: [
        {
          airline: {
            name: 'Vueling',
            code: 'VY',
            logoUrl: 'https://images.kiwi.com/airlines/64/VY.png',
            isLowCost: true
          },
          flightNumber: 8657,
          departure: {
            airport: {
              name: 'Václav Havel Airport Prague',
              city: { name: 'Prague' }
            }
          },
          arrival: {
            airport: { name: 'Barcelona–El Prat', city: { name: 'Barcelona' } }
          }
        }
      ],
      duration: 145,
      price: { amount: 46, currency: 'EUR' }
    }
  },
  {
    node: {
      id: 'RmxpZ2h0OjMzNzkyOTI4Nw==',
      departure: {
        time: '2018-03-06T13:35:00.000Z',
        localTime: '2018-03-06T14:35:00.000Z'
      },
      arrival: {
        time: '2018-03-06T16:00:00.000Z',
        localTime: '2018-03-06T17:00:00.000Z'
      },
      legs: [
        {
          airline: {
            name: 'Ryanair',
            code: 'FR',
            logoUrl: 'https://images.kiwi.com/airlines/64/FR.png',
            isLowCost: true
          },
          flightNumber: 3040,
          departure: {
            airport: {
              name: 'Václav Havel Airport Prague',
              city: { name: 'Prague' }
            }
          },
          arrival: {
            airport: { name: 'Barcelona–El Prat', city: { name: 'Barcelona' } }
          }
        }
      ],
      duration: 145,
      price: { amount: 65, currency: 'EUR' }
    }
  }
];

const pageInfo = {
  hasNextPage: true,
  hasPreviousPage: false,
  startCursor: 'YXJyYXljb25uZWN0aW9uOjA=',
  endCursor: 'YXJyYXljb25uZWN0aW9uOjQ='
};

describe('flights', () => {
  it('shows flights', () => {
    const query = {
      from: 'Prague',
      to: 'Barcelona',
      date: '2018-03-06'
    };
    const component = renderer.create(
      <Flights
        flights={flights}
        pageInfo={pageInfo}
        query={query}
        errors={[]}
      />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
