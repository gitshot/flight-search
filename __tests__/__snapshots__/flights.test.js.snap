// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`flights shows flights 1`] = `
<div
  className="Layout-s62o9bf-1 dIpton"
>
  <div
    className="Header-p2oiqz-1 fedUee"
  >
    <h1
      className="Title-s15je9jd-0 eRmqcx"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <strong>
          flight
        </strong>
        <br />
        search
      </a>
    </h1>
    <nav>
      <a
        className="Header__Anchor-p2oiqz-0 iPrusK"
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
      <a
        className="Header__Anchor-p2oiqz-0 iPrusK"
        href="/about"
        onClick={[Function]}
      >
        About
      </a>
    </nav>
  </div>
  <div
    className="Layout__Content-s62o9bf-0 fUQwJD"
  >
    <div
      className="Search-aeivv1-0 iYKKHW"
    >
      <form
        action="/flights"
      >
        <label>
          From:
          <div
            className="AutocompleteContainer-s1bbgvdj-0 jnlXpA"
          >
            <div
              style={
                Object {
                  "display": "inline-block",
                }
              }
            >
              <input
                aria-autocomplete="list"
                aria-expanded={false}
                autoComplete="off"
                id="from"
                name="from"
                onBlur={[Function]}
                onChange={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                role="combobox"
                value="Prague"
              />
            </div>
          </div>
        </label>
        <label>
          To:
          <div
            className="AutocompleteContainer-s1bbgvdj-0 jnlXpA"
          >
            <div
              style={
                Object {
                  "display": "inline-block",
                }
              }
            >
              <input
                aria-autocomplete="list"
                aria-expanded={false}
                autoComplete="off"
                id="to"
                name="to"
                onBlur={[Function]}
                onChange={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                role="combobox"
                value="Barcelona"
              />
            </div>
          </div>
        </label>
        <label>
          When:
          <div
            className="DatepickerContainer-s1jdkito-0 djxBjq"
          >
            <div>
              <div
                className="react-datepicker-wrapper"
              >
                <div
                  className="react-datepicker__input-container"
                >
                  <input
                    autoComplete={undefined}
                    autoFocus={undefined}
                    className=""
                    disabled={false}
                    id={undefined}
                    name={undefined}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    placeholder={undefined}
                    readOnly={undefined}
                    required={undefined}
                    tabIndex={undefined}
                    title={undefined}
                    type="text"
                    value="03/06/2018"
                  />
                </div>
              </div>
            </div>
            <input
              name="date"
              type="hidden"
              value="2018-03-06"
            />
          </div>
        </label>
        <input
          type="submit"
          value="Search"
        />
      </form>
    </div>
    <div
      className="flights__Content-s1wdw2le-0 kWmaRO"
    >
      <p>
        Searching for flights from 
        <strong>
          Prague
        </strong>
         to
         
        <strong>
          Barcelona
        </strong>
         on 
        <strong>
          2018-03-06
        </strong>
        .
      </p>
      <div
        className="SearchResult__ListItem-s1t0qfze-0 iBDcIE"
      >
        <div
          className="SearchResult__ListContent-s1t0qfze-1 dJKMAd"
        >
          <div
            className="SearchResult__ListItem__DateContainer-s1t0qfze-3 kUyWYw"
          >
            <div
              className="SearchResult__ListItem__Departure-s1t0qfze-4 bPYUZC"
            >
              <span>
                departure
              </span>
              21:10
            </div>
            <div
              className="SearchResult__ListItem__Arrival-s1t0qfze-5 hLJEMj"
            >
              <span>
                arrival
              </span>
              23:35
            </div>
          </div>
          <table>
            <tbody>
              <tr
                className="SearchResult__Leg-s1t0qfze-7 cWeHxz"
              >
                <td>
                  <img
                    src="https://images.kiwi.com/airlines/64/VY.png"
                    title="Vueling"
                  />
                </td>
                <td>
                  Václav Havel Airport Prague
                </td>
                <td>
                  Barcelona–El Prat
                </td>
              </tr>
            </tbody>
          </table>
          <div
            className="SearchResult__ListItemTitle-s1t0qfze-2 leiZA"
          >
            46
             
            EUR
          </div>
        </div>
        <div
          className="SearchResult__Footer-s1t0qfze-6 eesaqC"
        >
          <span>
            2
            h 
            25
            m
          </span>
        </div>
      </div>
      <div
        className="SearchResult__ListItem-s1t0qfze-0 iBDcIE"
      >
        <div
          className="SearchResult__ListContent-s1t0qfze-1 dJKMAd"
        >
          <div
            className="SearchResult__ListItem__DateContainer-s1t0qfze-3 kUyWYw"
          >
            <div
              className="SearchResult__ListItem__Departure-s1t0qfze-4 bPYUZC"
            >
              <span>
                departure
              </span>
              14:35
            </div>
            <div
              className="SearchResult__ListItem__Arrival-s1t0qfze-5 hLJEMj"
            >
              <span>
                arrival
              </span>
              17:00
            </div>
          </div>
          <table>
            <tbody>
              <tr
                className="SearchResult__Leg-s1t0qfze-7 cWeHxz"
              >
                <td>
                  <img
                    src="https://images.kiwi.com/airlines/64/FR.png"
                    title="Ryanair"
                  />
                </td>
                <td>
                  Václav Havel Airport Prague
                </td>
                <td>
                  Barcelona–El Prat
                </td>
              </tr>
            </tbody>
          </table>
          <div
            className="SearchResult__ListItemTitle-s1t0qfze-2 leiZA"
          >
            65
             
            EUR
          </div>
        </div>
        <div
          className="SearchResult__Footer-s1t0qfze-6 eesaqC"
        >
          <span>
            2
            h 
            25
            m
          </span>
        </div>
      </div>
      <div
        className="Pager-q4rar8-0 dzWOMs"
      >
        <a>
          Previous
        </a>
        <a
          href="/flights?from=Prague&to=Barcelona&date=2018-03-06&after=YXJyYXljb25uZWN0aW9uOjQ%3D&before="
          onClick={[Function]}
        >
          Next
        </a>
      </div>
    </div>
  </div>
</div>
`;
